### PR TITLE
Fix BN folding for PyTorch ConvTranspose2d with groups>1

### PR DESCRIPTION
**Pull request by elad-c**
## Pull Request Description:
Fix BN folding for PyTorch ConvTranspose2d with groups>1

## Checklist before requesting a review:
- [x] I set the appropriate labels on the pull request.
- [x] I have added/updated the release note draft (if necessary).
- [x] I have updated the documentation to reflect my changes (if necessary).
- [x] All function and files are well documented. 
- [x] All function and classes have type hints.
- [x] There is a licenses in all file.
- [x] The function and variable names are informative. 
- [x] I have checked for code duplications.
- [x] I have added new unittest (if necessary).

### CODE REVIEW

File: model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/batchnorm_folding.py [hunk 1]
```diff
@@ -66,7 +66,15 @@ def update_kernel_for_bn_folding_fn(conv_node: BaseNode,
         _scale = weights_scale[None, :, None, None]
     else:
         _scale = weights_scale[:, None, None, None]
-    return kernel * _scale, KERNEL
+    if conv_node.type == ConvTranspose2d and conv_node.framework_attr[GROUPS] > 1:
+        # PyTorch ConvTranspose2d kernel with groups stacks groups on in_channels axis, so need to reshape the kernel
+        # so the groups are stacked on the out_channels axis to match the scale vector (then reshape back to original
+        # shape)
+        _in_channels = int(conv_node.framework_attr[IN_CHANNELS]/conv_node.framework_attr['groups'])
+        _out_channels = conv_node.framework_attr[OUT_CHANNELS]
```
Comment (lior-dikstein) [inline model_compression_toolkit/core/pytorch/graph_substitutions/substitutions/batchnorm_folding.py:74]: please use constant instead of 'groups'

**Review by lior-dikstein — APPROVED** [review]
